core/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -403,6 +403,95 @@ class DeltaSuite extends QueryTest
     }
   }
 
+  test("batch write: append, dynamic partition overwrite") {
+    withTempDir { tempDir =>
+      def data: DataFrame = spark.read.format("delta").load(tempDir.toString)
+
+      Seq (1, 2, 3).toDF
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .partitionBy("part")
+        .mode("append")
+        .save(tempDir.getCanonicalPath)
+
+      Seq(1, 5).toDF
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .partitionBy("part")
+        .mode("overwrite")
+        .option("partitionOverwriteMode", "dynamic")
+        .save(tempDir.getCanonicalPath)
+      checkDatasetUnorderly(data.toDF.select($"value".as[Int]), 1, 2, 5)
+
+      Seq(("a", "x"), ("b", "y"), ("c", "x")).toDF("value", "part")
+        .write
+        .format("delta")
+        .partitionBy("part")
+        .mode("overwrite")
+        .option("overwriteSchema", true)
+        .save(tempDir.getCanonicalPath)
+
+      Seq(("a", "x"), ("d", "x")).toDF("value", "part")
+        .write
+        .format("delta")
+        .partitionBy("part")
+        .mode("overwrite")
+        .option("partitionOverwriteMode", "dynamic")
+        .save(tempDir.getCanonicalPath)
+      checkDatasetUnorderly(data.toDF.select($"value".as[String]), "a", "b", "d")
+    }
+  }
+
+  test("batch write: append, dynamic partition overwrite with replaceWhere") {
+    withTempDir { tempDir =>
+      def data: DataFrame = spark.read.format("delta").load(tempDir.toString)
+
+      Seq((1, "x"), (2, "y"), (3, "z")).toDF("value", "sub")
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .partitionBy("part", "sub")
+        .mode("append")
+        .save(tempDir.getCanonicalPath)
+
+      Seq((3, "x"), (5, "x")).toDF("value", "sub")
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .partitionBy("part", "sub")
+        .mode("overwrite")
+        .option("replaceWhere", "part==1")
```

Comment:
1. The entire suite uses the DeltaOptions key
2. The condition is invalid (only one equal instead of two)
```suggestion
        .option(DeltaOptions.REPLACE_WHERE_OPTION, "part = 1")
```

Reply:
interesting. i never had any issues with two equal signs, it seems to work as well. i find the one equal sign off-putting, it looks like assignment, but ok if thats the standard i will change it